src/adagrams.js

```diff
@@ -1,15 +1,148 @@
+  const LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  let lettersList = [];
+  let letterFreq = {};
+
+
+    while (lettersList.length < 10) {
+      let randomLetter = Object.keys(LETTER_POOL)[Math.floor(Math.random() * Object.keys(LETTER_POOL).length)];
+
+    if (randomLetter in letterFreq) {
+      letterFreq[randomLetter] += 1;
+
+    if (letterFreq[randomLetter] < LETTER_POOL[randomLetter]) {
+      lettersList.push(randomLetter);
+    }
+  } else {
+    letterFreq[randomLetter] = 1;
+    lettersList.push(randomLetter)
+  }
+}
+
+return lettersList;
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
 };
 
+export const usesAvailableLetters = (word, letterBank) => {
+
+  let bankCopy = letterBank.slice();
+  let wordUpper = word.toUpperCase();
+  
+    for (let i = 0; i < wordUpper.length; i++) {
```

Comment:
We could also use a for loop that runs through each letter of the word so we don't have to worry about indexing!

     for (let letter of wordUpper){
          ...